lib_ux_sync/src/ux_sync.c

```diff
@@ -0,0 +1,162 @@
+#ifdef HAVE_NBGL
+
+#include "ux_sync.h"
+
+static ux_sync_ret_t g_ret;
+static bool          g_ended;
+
+static void choice_callback(bool confirm)
+{
+    if (confirm) {
+        g_ret = UX_SYNC_RET_APPROVED;
+    }
+    else {
+        g_ret = UX_SYNC_RET_REJECTED;
+    }
+
+    g_ended = true;
+}
+
+static void quit_callback(void)
+{
+    g_ret   = UX_SYNC_RET_QUITTED;
+    g_ended = true;
+}
+
+static void rejected_callback(void)
+{
+    g_ret   = UX_SYNC_RET_REJECTED;
+    g_ended = true;
+}
+
+static void ux_sync_init(void)
+{
+    g_ended = false;
+    g_ret   = UX_SYNC_RET_ERROR;
+}
+
+static ux_sync_ret_t ux_sync_wait(bool exitOnApdu)
+{
+    bool apduReceived;
+
+    while (!g_ended) {
+        apduReceived = io_recv_and_process_event();
+        if (exitOnApdu && apduReceived) {
+            return UX_SYNC_RET_APDU_RECEIVED;
+        }
+    }
+
+    return g_ret;
+}
+
+ux_sync_ret_t ux_sync_homeAndSettings(const char                   *appName,
```

Comment:
Description of each function is missing

Reply:
Done, I hope someone will use it before the comment becomes misleading as every unused doc...